src/builtin.c

```diff
@@ -1201,9 +1201,21 @@ static jv f_string_indexes(jq_state *jq, jv a, jv b) {
 }
 
 static jv f_string_implode(jq_state *jq, jv a) {
+  int i;
+  int len;
   if (jv_get_kind(a) != JV_KIND_ARRAY) {
     return ret_error(a, jv_string("implode input must be an array"));
   }
+  len = jv_array_length(jv_copy(a));
+  for (i = 0; i < len; i++) {
+    jv n = jv_array_get(jv_copy(a), i);
+    if (jv_get_kind(n) != JV_KIND_NUMBER || jvp_number_is_nan(n)) {
+      jv_free(a);
+      return type_error(n, "codepoint must be a number");
+    }
+    jv_free(n);
```

Comment:
Is there some nicer way to do this? i peek around in the rest of the code and it seems to be what you have to do if you want to iterate thru element but not use them for something.

I also noticed this possibly leak in `TO_TM_FIELD` https://github.com/jqlang/jq/blob/master/src/builtin.c#L1468 wont `n` leak here?

Reply:
> I also noticed this possibly leak in `TO_TM_FIELD` https://github.com/jqlang/jq/blob/master/src/builtin.c#L1468 wont `n` leak here?

Yes! That's a leak.

```bash
$ ./jq -cn '["1",2,3,4,5,6]|mktime'
jq: error (at <unknown>): mktime requires parsed datetime inputs

=================================================================
==862970==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 49 byte(s) in 1 object(s) allocated from:
    #0 0x7f40282e1359 in __interceptor_malloc /usr/src/debug/gcc/gcc/libsanitizer/asan/asan_malloc_linux.cpp:69
    #1 0x564963421c58 in jv_mem_alloc src/jv_alloc.c:122

SUMMARY: AddressSanitizer: 49 byte(s) leaked in 1 allocation(s).
```

Reply:
https://github.com/jqlang/jq/pull/2763

Reply:
Yeap would like to avoid iterate twice. Looks like the builtins usually do error checking and not much logic? i started to move error checking into `jv_string_implode` ... but to get nice error message should `type_error` move to jv.c etc? hmm. feels a bit ugly to move everything into `f_implode`?

Reply:
I think just copy the body of `jv_string_implode()` into `f_implode()` and suffer the code duplication.

Reply:
Something like that? `jv_string_implode` has to stay as it's part of the public API?

Reply:
That's my thinking.  I mean, we don't commit to the stability of the `jv`/`jq` APIs at all, but if we're going to break backwards-compatibility we should at least bump the SOVERSION.  In this case I am willing to suffer the code bloat.